### DISCUSSION TITLE
Battery Protect Funktion

### DISCUSSION THREAD

**Discussion by braini75** [tbnobody/OpenDTU#1404] [Ideas]
Wenn man einen OpentDTU kompatiblen Wechselrichter zur (Nacht)Einspeisung via Akku benutzt, möchte (und sollte) man eine Tiefentladung der Batterie vermeiden. Dazu gibt es verschiedene Methoden, die sich auch (mehr oder weniger) bewährt haben. Oft wir dazu die DC Seite am Wechselrichter getrennt, wenn die Batterie unter eine gewisse Spannung fällt (also entladen ist). Dadurch wird ein weiteres entladen verhindert. 
Der große Nachteil dabei ist, dass beim erneuten Verbinden der DC Seite mit der Batterie riesige Ströme (>100A) auftreten, da die Wechselrichter sehr große Transistoren besitzen, welche sich aufladen. Dadurch wird ein extra Strombegrenzer oder eine vorgeschaltet "Anfahrspannung" benötigt. Normalerweise würde am DC Eingang ein Solarpanel erwartet, welches nach und nach Leistung in den Wechselrichter abgibt und so die Kondensatoren langsam aufladen.

Eine weitere Möglichkeit, die ich derzeit sehr erfolgreich über Home Assistant nutze:
- prüfe die Eingangsspannung am PV Eingang
- schalte Inverter via openDTU aus, wenn am DC Eingang 23,5V unterschritten werden ( ich benutze 24V Blei Akkus)
- schalte Inverter wieder ein, wenn die DC Spannung > 27 Volt beträgt

So bleibt die DC Seite permanent verbunden und verhindert ein Entladen der Kondensatoren. Dadurch wird der Einschaltstrom nicht unnötig hoch, wenn die Batterie wieder zugeschaltet wird.

Ein Wunsch wäre eine optionale Batteriesteuerung im OpenDTU, welches die Eingabe einer Abschaltspannung und einer Einschaltspannung ermöglicht und den Inverter bei Unterschreiten der Abschaltspannung abschaltet und bei Überschreiten der Einschaltspannung wieder anschaltet.

**Comment by Head** [comment]
Vielleicht bist du da hier besser bedient? https://github.com/helgeerbe/OpenDTU-OnBattery
